pkg/splunk/enterprise/clustermaster.go

```diff
@@ -48,6 +48,7 @@ func ApplyClusterMaster(ctx context.Context, client splcommon.ControllerClient,
 	reqLogger := log.FromContext(ctx)
 	scopedLog := reqLogger.WithName("ApplyClusterMaster")
 	eventPublisher, _ := newK8EventPublisher(client, cr)
+	cr.Kind = "ClusterMaster"
```

Comment:
Same here?, and other places as well. Just curious why we are setting it explicitly, Isn't it the basic info, that should have been there alrady by the time we come here?

Reply:
Mostly because when the mock test runs , its not setting `Kind` field in the structure. just to. make sure Unit test case works fine. but this is automatically set in run time

Reply:
Same as above & other places.

Reply:
I will create ticket to fix this issue after the release.